vulkancapsviewer.cpp

```diff
@@ -655,22 +655,38 @@ bool vulkanCapsViewer::initVulkan()
 
 #if defined(VK_USE_PLATFORM_WAYLAND_KHR)
         if (surface_extension == VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME) {
-            VkWaylandSurfaceCreateInfoKHR surfaceCreateInfo = {};
-            surfaceCreateInfo.pNext = nullptr;
-            surfaceCreateInfo.sType = VK_STRUCTURE_TYPE_WAYLAND_SURFACE_CREATE_INFO_KHR;
-            surfaceCreateInfo.display = wl_display_connect(NULL);
-            surfaceCreateInfo.surface = nullptr;
-            surfaceResult = vkCreateWaylandSurfaceKHR(vkInstance, &surfaceCreateInfo, nullptr, &surface);
+            struct wl_display* display = wl_display_connect(NULL);
+            if (display != NULL) {
+                VkWaylandSurfaceCreateInfoKHR surfaceCreateInfo = {
+                  .sType = VK_STRUCTURE_TYPE_WAYLAND_SURFACE_CREATE_INFO_KHR,
+                  .pNext = nullptr,
+                  .flags = 0,
+                  .display = display,
+                  .surface = nullptr

```

Comment:
It just works like this ?

The reference states that the display and surface *must* be valid pointers to wayland objects.
https://www.khronos.org/registry/vulkan/specs/1.2-khr-extensions/html/chap31.html#VUID-VkWaylandSurfaceCreateInfoKHR-surface-01305

Reply:
Okay, just checked myself, and even the validation layers don't say anything. Is it a bug ?

Reply:
Whether it's a bug or not, exactly the same solution was in the code before, it's not something this patch adds. Seems getting legit wl_surface requires using QPA: https://bugreports.qt.io/browse/QTBUG-76983